anyway/models.py

```diff
@@ -871,13 +872,52 @@ class NewsFlash(Base):
     street2_hebrew = Column(Text(), nullable=True)
     non_urban_intersection_hebrew = Column(Text(), nullable=True)
     road_segment_name = Column(Text(), nullable=True)
+    critical = Column(Boolean(), nullable=True)
     newsflash_location_qualification = Column(
         Integer(),
         nullable=False,
         server_default=text(f"{NewsflashLocationQualification.NOT_VERIFIED.value}"),
     )
     location_qualifying_user = Column(BigInteger(), nullable=True)
 
+    def __init__(self, **kwargs):
+        super(NewsFlash, self).__init__(**kwargs)
+        self.critical = self.get_critical()
+
+    def get_critical(
+        self,
+        years_before=5,
+        suburban_road_severe_value=10,
+        suburban_road_killed_value=3,
+        urban_severe_value=2,
+    ):
+        from anyway.widgets.suburban_widgets.injured_count_by_severity_widget import (
+            InjuredCountBySeverityWidget,
+        )
+
+        if self.road1 is None or self.road_segment_name is None or self.date is None:
+            return None
+        resolution = BE_CONST.ResolutionCategories(self.resolution)
```

Comment:
I think it is worthwhile to handle the situation where the value in news_flash is not a legal `resolution`.

Reply:
i init the critical field as None so if  type of the resolution, not SUBURBAN_ROAD or STREET 
i return None.
it this cover this case

Reply:
I am not sure. Maybe it is worthwhile to check in the news-flash table whether in these criteria the resolution is always legal.

Reply:
> i init the critical field as None so if type of the resolution, not SUBURBAN_ROAD or STREET i return None. it this cover this case?

Note that the code that is used to calculate the values works only for STREET resolutions. If we need to calculate also for SUBURBAN_ROAD, we need to take the calculation from a different widget.